exonum-java-binding-qa-service/src/main/java/com/exonum/binding/qaservice/transactions/QaTransactionConverter.java

```diff
@@ -36,29 +36,29 @@
 public final class QaTransactionConverter implements TransactionConverter {
 
   @VisibleForTesting
-  static final ImmutableMap<Short, Function<BinaryMessage, Transaction>> TRANSACTION_FACTORIES =
-      ImmutableMap.<Short, Function<BinaryMessage, Transaction>>builder()
-          .put(INCREMENT_COUNTER.id(), IncrementCounterTx.converter()::fromMessage)
-          .put(CREATE_COUNTER.id(), CreateCounterTx.converter()::fromMessage)
-          .put(INVALID.id(), InvalidTx.converter()::fromMessage)
-          .put(INVALID_THROWING.id(), InvalidThrowingTx.converter()::fromMessage)
-          .put(VALID_THROWING.id(), ValidThrowingTx.converter()::fromMessage)
-          .put(VALID_ERROR.id(), ValidErrorTx.converter()::fromMessage)
+  static final ImmutableMap<Short, Function<RawTransaction, Transaction>> TRANSACTION_FACTORIES =
+      ImmutableMap.<Short, Function<RawTransaction, Transaction>>builder()
+          .put(INCREMENT_COUNTER.id(), IncrementCounterTx.converter()::fromRawTransaction)
+          .put(CREATE_COUNTER.id(), CreateCounterTx.converter()::fromRawTransaction)
+          .put(INVALID.id(), InvalidTx.converter()::fromRawTransaction)
+          .put(INVALID_THROWING.id(), InvalidThrowingTx.converter()::fromRawTransaction)
+          .put(VALID_THROWING.id(), ValidThrowingTx.converter()::fromRawTransaction)
+          .put(VALID_ERROR.id(), ValidErrorTx.converter()::fromRawTransaction)
           .build();
 
   @Override
-  public Transaction toTransaction(BinaryMessage message) {
-    checkServiceId(message);
+  public Transaction toTransaction(RawTransaction rawTransaction) {
+    checkServiceId(rawTransaction);
 
-    short txId = message.getMessageType();
+    short txId = rawTransaction.getTransactionId();
     return TRANSACTION_FACTORIES.getOrDefault(txId, (m) -> {
       throw new IllegalArgumentException("Unknown transaction id: " + txId);
     })
-        .apply(message);
+        .apply(rawTransaction);
   }
```

Comment:
possibly, can replace this method with one, extracted from `com.exonum.binding.qaservice.transactions.TransactionPreconditions.checkTransaction`